nixos/modules/services/networking/firezone/server.nix

```diff
@@ -0,0 +1,1210 @@
+      description = ''
+        Environment variables for this component of the Firezone server. For a
+        list of available variables, please refer to the [upstream definitions](https://github.com/firezone/firezone/blob/main/elixir/apps/domain/lib/domain/config/definitions.ex).
+        Some variables like `OUTBOUND_EMAIL_ADAPTER_OPTS` require json values
+        for which you can use `VAR = builtins.toJSON { /* ... */ }`.
+
+        This component will automatically inherit all variables defined via
+        {option}`services.firezone.server.settings` and
+        {option}`services.firezone.server.settingsSecret`, but which can be
+        overwritten by this option.
+      '';
+      default = { };
+      type = types.submodule {
+        freeformType = types.attrsOf (
+          types.oneOf [
+            types.bool
+            types.float
+            types.int
+            types.str
+            types.path
+            types.package
+          ]
+        );
+      };
+    };
+  };
+in
+{
+  options.services.firezone.server = {
+    enable = mkEnableOption "all Firezone components";
+    enableLocalDB = mkEnableOption "a local postgresql database for Firezone";
+    nginx.enable = mkEnableOption "nginx virtualhost definition";
+
+    openClusterFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Opens up the erlang distribution port of all enabled components to
+        allow reaching the server cluster from the internet. You only need to
+        set this if you are actually distributing your cluster across multiple
+        machines.
+      '';
+    };
+
+    clusterHosts = mkOption {
+      type = types.listOf types.str;
+      default = [
+        "api@localhost.localdomain"
+        "web@localhost.localdomain"
+        "domain@localhost.localdomain"
+      ];
+      description = ''
+        A list of components and their hosts that are part of this cluster. For
+        a single-machine setup, the default value will be sufficient. This
+        value will automatically set `ERLANG_CLUSTER_ADAPTER_CONFIG`.
+
+        The format is `<COMPONENT_NAME>@<HOSTNAME>`.
+      '';
+    };
+
+    settingsSecret = mkOption {
+      default = { };
+      description = ''
+        This is a convenience option which allows you to set secret values for
+        environment variables by specifying a file which will contain the value
+        at runtime. Before starting the server, the content of each file will
+        be loaded into the respective environment variable.
+
+        Otherwise, this option is equivalent to
+        {option}`services.firezone.server.settings`. Refer to the settings
+        option for more information regarding the actual variables and how
+        filtering rules are applied for each component.
+      '';
+      type = types.submodule {
+        freeformType = types.attrsOf types.path;
+        options = {
+          RELEASE_COOKIE = mkOption {
```

Comment:
Since these are paths, it would be clearer to suffix with `_FILE`. Or are they directly read by firezone itself?

Reply:
The variables in here will be directly read by firezone, but firezone expects them to contain the value, not a file. But since there are so many variables that a user may want/need to pass as a secret, I opted for a `settingsSecret` as a pendant to `settings`, which just reads the content from a file and sets the variable.. I find this better than giving the user a `environmentFile` option, since its allows us to document the variables which are being set and we can use `LoadCredential` in the background. So basically "RFC42 style secret settings", hope that makes sense